### ISSUE TITLE
change data in a FitPage (Trac #1185)

### ISSUE THREAD

**Issue by RichardHeenan** [SasView/sasview#1221]
It still annoys me greatly that having set up a complicated model in a !FitPage you have to start from scratch to reuse it with a different data set. Thus is particularly important if you have constraints between two or three !FitPages.

I would like to send a fresh data set to a !FitPage, at the same time "freezing" any fit results currently there so they are still available.

Is there any reason why this would not work? (Also is there a ticket for this somewhere already?)

Migrated from http://trac.sasview.org/ticket/1185
```json
{
    "status": "new",
    "changetime": "2018-10-08T09:10:25",
    "_ts": "2018-10-08 09:10:25.812689+00:00",
    "description": "It still annoys me greatly that having set up a complicated model in a !FitPage you have to start from scratch to reuse it with a different data set. Thus is particularly important if you have constraints between two or three !FitPages.\n\nI would like to send a fresh data set to a !FitPage, at the same time \"freezing\" any fit results currently there so they are still available.\n\nIs there any reason why this would not work? (Also is there a ticket for this somewhere already?)",
    "reporter": "richardh",
    "cc": "",
    "resolution": "",
    "workpackage": "SasView GUI Enhancements",
    "time": "2018-09-28T17:06:47",
    "component": "SasView",
    "summary": "change data in a FitPage",
    "priority": "major",
    "keywords": "",
    "milestone": "SasView 5.1.0",
    "owner": "richardh",
    "type": "enhancement"
}
```

**Comment by butlerpd** [comment]
Trac update at `2018/10/06 16:22:27`: **butler** commented:

I think this issue is the subject of another ticket by @richardh but not sure what to search for.  If we cannot find it may just proceed and eventually close the other as duplicate, if and when it gets found?

Question 1: currently one can change the model used on a !FitPage but I don't think we can currently send a new data set to a current !FitPage can we?  If one sends a new data set to fitting I believe a new !FitPage is created and the data sent there.  In this case one would have to re-select the PQ and, if necessary, the S(Q), but then one should be able to use "copy paste parameters" to set up the same model?  We used to have problems with not all parameters being saved but thought that was fixed at some point?

Also, batch fitting provides a way (wont work with complex constraints between tabs of course) to fit multiple data to the same arbitrarily complex model.

Perhaps it would help to have a bit more detailed description of the use case? Or is the use case strictly the constrained fitting where the constraints will be defined against a particular set of !FitPages?

Question 2: Is the suggestion to allow sending new data to a !FitPage that **already** contains data (and a model with parameters etc set)?  There is currently a label box that contains the data name that I suppose could be turned into a drop down box to choose any data set in the datamanager, and I think that is done in some panel (maybe batch fit?), but I think that could cause a lot of user confusion? I guess I am not entirely clear on what is being suggested here?

**Comment by butlerpd** [comment]
Trac update at `2018/10/06 16:25:35`: **butler** changed description from:

> It still annoys me greatly that having set up a complicated model in a FitPage you have to start from scratch to reuse it with a different data set. Thus is particulalry important if you have constraints between two or threee FitPages.
> 
> I would like to send a fresh data set to a FitPage, at the same time "freezing" any fit results currently there so they are still available.
> 
> Is there any reason why this would not work? (Also is there a ticket for this somewhere already?)

to:

> It still annoys me greatly that having set up a complicated model in a !FitPage you have to start from scratch to reuse it with a different data set. Thus is particularly important if you have constraints between two or three !FitPages.
> 
> I would like to send a fresh data set to a !FitPage, at the same time "freezing" any fit results currently there so they are still available.
> 
> Is there any reason why this would not work? (Also is there a ticket for this somewhere already?)

**Comment by RichardHeenan** [comment]
Trac update at `2018/10/08 09:10:25`: **richardh** commented:

Re Q1, correct, we can change model but not the data on a given fit page. As noted in Q2 in sasview 4.x the "data" name at top of page looks like a drop down selector, but disabled.

Re Q2 yes I do want to send new data to an existing fit page that already contains some data. Ideally sasview could automatically freeze any fit results already associated with that fit page. (Note in 5.0 freeze is now adding a unique time stamp to the end of the workspace names.]

Re copy/paste - it is still tedious to have to send data to a new fit page, find the correct P(Q), S(Q), then use edit/copy and edit/paste to set up the parameters, then set up any inter-fit page constraints. [Need to check in 5.0 that copy & paste does actually copy ALL the other settings & intra fit page constraints.]

For complex models & constraints it would be very useful to be able to save project, perhaps send it to another user, then be able to pull different data into it without any hassle. At present we are always locked in to the original data sets.

**Comment by RichardHeenan** [comment]
Addressed at last by #1537 so closing this issue!
